test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -2017,4 +2025,75 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthFo
       });
     });
   });
-});
+  
+  Expect.describe('Registration Flow', function () {
+    itp('does not show the registration button if settings.registration is not set', function () {
+      return setup().then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(0);
+      });
+    });
+    itp('does not show the registration button if settings.registration is empty object', function () {
+      var registration =  {};
+      return setupRegistrationButton(registration).then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(0);
+      });
+    });
+    itp('does not show the registration button if settings.registration properties are undefined', function () {
+      var registration =  undefined;
+      return setupRegistrationButton(registration).then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(0);
+      });
+    });
+    itp('shows the registration button if settings.registration properties are configured', function () {
+      var registration =  {
+        click: function () {
+          window.location.href = 'http://www.test.com';
+        },
+        label: 'Dont have an account ?',
+        text: 'Register'
+      };
+      
+      spyOn(Okta, 'loc').and.callFake(function (key) {
+        if (key === 'registration.signup.label') {
+          return 'Dont have an account ?';
+        } else if (key === 'registration.signup.text') {
+          return 'Register';
+        }
+      });
+      return setupRegistrationButton(registration).then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(1);
+        expect(test.form.registrationLabel().length).toBe(1);
+        expect(test.form.registrationLabel().text()).toBe('Dont have an account ?');
+        expect(test.form.registrationLink().length).toBe(1);
+        expect(test.form.registrationLink().text()).toBe('Register');
+        expect(typeof(registration.click)).toEqual('function');
+      });
+    });
+    itp('calls settings.registration.click if its a function and when the link is clicked', function () {
+      var registration =  {
+        click: function () {
+          window.location.href = 'http://www.test.com';
+        },
+        label: 'Dont have an account ?',
+        text: 'Register'
+      };
+      spyOn(registration, 'click');
+      return setupRegistrationButton(registration).then(function (test) {
+        expect(typeof(registration.click)).toEqual('function');
```

Comment:
What is this expect for? The test defines registration.click a couple lines above.